### ISSUE TITLE
the variable X_test  is never used in whole program ?

### ISSUE THREAD

**Issue by preesee** [eliorc/Medium#2]
Is the variable X_test  never consumed in whole program ? What is this variable's purpose ?

**Comment by eliorc** [comment]
Nope, when training models you keep 3 datasets, training validation and test. The validation is what you use to monitor the training process therefore it is kind of prone to overfitting, as you are playing with hyperparameters with the aim of maxing out performance on validation set.

The test set is never tuned against, and is used to compare different models objectively, so I kept here a test set so if you train other models against it it would be comparable.

I am only showing the training process here so no, the test set wasn't used
